crates/runtime/src/dataaccelerator/metadata.rs

```diff
@@ -16,9 +16,10 @@ limitations under the License.
 
 //! Store and retrieve Spice metadata in durable accelerator tables
 //!
-//! The metadata table will be a new table `spice_sys_metadata` with two columns:
+//! The metadata table will be a new table `spice_sys_metadata` with three columns:
 //! - `dataset` (PRIMARY KEY, TEXT): The dataset the metadata entry corresponds to
-//! - `metadata` (TEXT): The metadata entry in JSON format
+//! - `key` (PRIMARY KEY, TEXT): The key of the metadata entry
```

Comment:
I would add this to the explanation for `key`, _"allow multiple metadata values to be associated with a dataset. "_.